addons/survey/static/src/js/survey_form.js

```diff
@@ -0,0 +1,272 @@
+var field_utils = require('web.field_utils');
+var rpc = require('web.rpc');
+var publicWidget = require('web.public.widget');
+var time = require('web.time');
+
+publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend({
+    selector: '.o_survey_form',
+    events: {
+        'change .o_survey_form_simple_radio input[type="radio"]': '_onChangeRadio',
+        'change .o_survey_form_select': '_onChangeSelect',
+        'change .o_survey_form_checkbox_other': '_onChangeMultipleCheckbox',
+        'click button[name="button_submit"]': '_onSubmit',
+    },
+
+    //--------------------------------------------------------------------------
+    // Widget
+    //--------------------------------------------------------------------------
+
+    /**
+    * @override
+    */
+    willStart: function () {
+        var url = '/web/webclient/locale/' + (document.documentElement.getAttribute('lang') || 'en_US').replace('-', '_');
+        var localeReady = ajax.loadJS(url);
+        return Promise.all([this._super.apply(this, arguments), localeReady]);
+    },
+
+    /**
+    * @override
+    */
+    start: function () {
+        var self = this;
+        self.surveyToken = self.$target.data('surveyToken');
+        self.answerToken = self.$target.data('answerToken');
+
+        var $timer = $('.o_survey_timer');
+        if ($timer.length) {
+            self.surveyTimerWidget = new publicWidget.registry.SurveyTimerWidget(null, $timer.data());
+            self.surveyTimerWidget.attachTo($timer);
+            self.surveyTimerWidget.on('time_up', self, function (ev) {
+                self.$target.find('button[name="button_submit"]').click();
+            });
+        }
+
+        _.each(this.$('div.o_survey_form_date'), function (date_group) {
+            self._updateDateForDisplay($(date_group));
+        });
+
+        _.each(this.$('div.o_survey_form_simple_radio'), function (radio_group) {
+            self._updateSimpleRadioForDisplay($(radio_group));
+        });
+
+        _.each(this.$('div.o_survey_form_simple_select'), function (select_group) {
+            self._updateSimpleSelectForDisplay($(select_group));
+        });
+
+        return this._super.apply(this, arguments);
+    },
+
+    // -------------------------------------------------------------------------
+    // Private
+    // -------------------------------------------------------------------------
+
+    // EVENTS
+    // -------------------------------------------------------------------------
+    _onChangeRadio: function (event) {
+        var $radio_group = $(event.currentTarget).parents('.o_survey_form_simple_radio');
+        return this._updateSimpleRadioForDisplay($radio_group);
+    },
+
+    _onChangeSelect: function (event) {
+        var $select_group = $(event.currentTarget).parents('.o_survey_form_simple_select');
+        return this._updateSimpleSelectForDisplay($select_group);
+    },
+
+    _onChangeMultipleCheckbox: function (event) {
```

Comment:
comment have been added on the called method inside this one. Could not really explain more here, except, that on checkbox change, it will call the submethod :)